Mantis #30823: add Course AboStatus to XML export & import (ILIAS10)

GitHamo opened the pull request:
Sister PR https://github.com/ILIAS-eLearning/ILIAS/pull/10592

GitHamo committed:
add Course AboStatus to XML export & import

sKarki999 commented:
LGTM :+1:  
Thanks!

schmitz-ilias commented:
Hi @GitHamo, again sorry for the wait and thanks for this fix!

a few things before we can merge this:

- Please remove the change to `ilias_crs_9_0.xsd`, as this fix won't make it down to 9.
- In `ilias_crs_10_0.xsd` please change `minOccurs` of `AboStatus` to `0`, so that existing exports from ILIAS 10 can still be imported.
- There is a bespoke schema specifically for the administration of courses via SOAP, `ilias_ws_crs_<version>.xsd`. Please also add `AboStatus` there as an optional element. Since `ilias_ws_crs_10_0.xsd` doesn't exist, adding this in 11 and trunk only is fine.

Please also change #10662 and #11074 accordingly. If you have any questions, let me know. Thank you!

Best, Tim

GitHamo commented:
Hi @schmitz-ilias,

Thanks for the review and the feedback.

I have applied the requested changes:

* removed the change to `ilias_crs_9_0.xsd`
* changed `AboStatus` to optional (`minOccurs="0"`) in `ilias_crs_10_0.xsd`
* added `AboStatus` as an optional element to `components/ILIAS/Export/xml/SchemaValidation/ilias_ws_crs_11_0.xsd`.

I have also applied the corresponding version-specific changes in the other branches/PRs (#10662 and #11074) and cherry-picked the commits accordingly.

Please let me know if you have any comments.

Thanks again!

schmitz-ilias commented:
Looks good to me, thank you @GitHamo!

I'll merge this and all sister PRs.
